api/src/main/java/org/openmrs/module/eptsreports/reporting/library/queries/TxMlQueries.java

```diff
@@ -397,24 +487,67 @@ public static String getPatientsTracedWithVisitCard(
       int arvPediatriaSeguimentoEncounterTypeId,
       int returnVisitDateForDrugsConcept,
       int returnVisitDateConcept,
+      int masterCardDrugPickupEncounterTypeId,
       int homeVisitCardEncounterTypeId,
       int apoioReintegracaoParteAEncounterTypeId,
       int apoioReintegracaoParteBEncounterTypeId,
       int typeOfVisitConcept,
       int buscaConcept,
       int patientFoundConcept,
-      int patientFoundAnswerConcept) {
+      int patientFoundAnswerConcept,
+      int artDatePickupConceptId) {
     String query =
         "SELECT DISTINCT pa.patient_id FROM patient pa "
-            + "    INNER JOIN ("
-            + "        SELECT p.patient_id,MAX(e.encounter_datetime) return_date FROM patient p "
-            + "            INNER JOIN encounter e ON e.patient_id = p.patient_id AND e.encounter_datetime <= :endDate AND e.location_id=:location "
-            + "            INNER JOIN obs o ON o.encounter_id = e.encounter_id AND o.location_id= :location "
-            + "        WHERE p.voided = 0 AND e.voided = 0 AND o.voided=0 "
-            + "            AND e.encounter_type IN (%d,%d,%d) "
-            + "            AND o.concept_id IN (%d, %d) "
-            + "            AND e.location_id = :location "
-            + "        GROUP BY p.patient_id)lp ON pa.patient_id=lp.patient_id "
+            + "    INNER JOIN ( "
+            + "    SELECT patient_id, MAX(return_date) AS return_date "
+            + "    FROM (SELECT p.patient_id, "
+            + "                 (SELECT o.value_datetime AS return_date "
```

Comment:
the spec says the most recent scheduled date, on my understanding it looks like this query  is picking  most recent encounter

Reply:
@cfaife please look again:

> The most recent scheduled date is the most recent one from the following 3 sources:
> 
> (1) last scheduled drug pick up (Concept Id 5096) date from last Fila form (Encounter Type 18, encounter_datetime <= endDate) and the 
(2) last scheduled consultation date (Concept Id 1410) from last Ficha Seguimento or Ficha Clinica Form (Encounter Type 6 or 9, encounter_datetime <= endDate) and 
(3) 30 days after the last ART pickup date (Concept Id 23866) from last Recepcao – Levantou ARV Form (Encounter Type 52, encounter_datetime <= endDate)

If it is not following the spec, what should be changed?

Reply:
now I see